src/wizard.ts

```diff
@@ -0,0 +1,85 @@
+import fs from 'fs'
+import prompt from 'prompts'
+
+interface ProjectOptions {
+  projectName: string
+  documentSyntax: string
+  entryPoint: string
+}
+
+export const getProjectNamePrompt = (
+  processDirectory: string,
+): prompt.PromptObject => ({
+  type: 'text',
+  name: 'projectName',
+  message: 'Project name',
+  initial: processDirectory,
+  validate: (input) =>
+    !!input.match(/^[\w\-_]+$/) ||
+    'Project name may only contain letters, hyphens and underscores',
+})
+
+export const getDocumentSyntaxPrompt = (): prompt.PromptObject => ({
+  name: 'documentSyntax',
+  type: 'select',
+  message: 'Document syntax',
+  choices: [
+    { title: 'json', value: 'json' },
+    { title: 'yaml', value: 'yaml' },
+  ],
+  initial: 0,
+})
+
+export const getEntryPointPrompt = (
+  documentSyntax: string,
+): prompt.PromptObject => ({
+  name: 'entryPoint',
+  type: 'text',
+  message: 'Entry point',
+  initial: `openapi.${documentSyntax}`,
+  validate: (input) =>
+    !!input.match(
+      new RegExp(
+        `(.*)(?<!\\.|${documentSyntax === 'json' ? 'yaml' : 'json'})$`,
+      ),
+    ) ||
+    `Entry point extension must match chosen document syntax (you chose ${documentSyntax})`,
+  format: (input): string => {
+    if (input.match(/(\.json|\.yaml)$/)) {
```

Comment:
Consider case sensitivity here, some people put their extensions in uppercase.

Reply:
added an `i`